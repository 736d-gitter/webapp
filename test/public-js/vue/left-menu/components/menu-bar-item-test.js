'use strict';

const { createLocalVue, shallowMount } = require('@vue/test-utils');
const Vuex = require('vuex');

const createStore = require('../../../../../public/js/vue/store').default;
const actions = require('../../../../../public/js/vue/store/actions');
const MenuBarItem = require('../../../../../public/js/vue/left-menu/components/menu-bar-item.vue');

let wrapper;
let stubbedActions = {};
function factory(propsData = {}, extendStore = () => {}) {
  const localVue = createLocalVue();
  localVue.use(Vuex);

  Object.keys(actions).forEach(actionKey => {
    stubbedActions[actionKey] = jest.fn();
  });

  const store = createStore({
    actions: stubbedActions
  });
  extendStore(store);

  wrapper = shallowMount(MenuBarItem.default, {
    localVue,
    store,
    propsData
  });
}

describe('menu-bar-item', () => {
  afterEach(() => {
    wrapper.destroy();
  });

  it('all matches snapshot', () => {
    factory({
      type: 'all'
    });
    expect(wrapper.element).toMatchSnapshot();
  });

  it('when all state, all item has highlight matches snapshot', () => {
    factory(
      {
        type: 'all'
      },
      store => {
        store.state.leftMenuState = 'all';
      }
    );
    expect(wrapper.element).toMatchSnapshot();
  });

  it('search matches snapshot', () => {
    factory({
      type: 'search'
    });
    expect(wrapper.element).toMatchSnapshot();
  });

  it('people matches snapshot', () => {
    factory({
      type: 'people'
    });
    expect(wrapper.element).toMatchSnapshot();
  });

  it('create matches snapshot', () => {
    factory({
      type: 'create'
    });
    expect(wrapper.element).toMatchSnapshot();
  });

  it('toggle matches snapshot', () => {
    factory({
      type: 'toggle'
    });
    expect(wrapper.element).toMatchSnapshot();
  });
});
