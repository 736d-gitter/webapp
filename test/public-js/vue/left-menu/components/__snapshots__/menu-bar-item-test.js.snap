// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`menu-bar-item all matches snapshot 1`] = `
<button
  class="item item-all active"
  type="button"
>
  <span
    class="icon-wrapper"
  />
</button>
`;

exports[`menu-bar-item create matches snapshot 1`] = `
<button
  class="item item-create"
  type="button"
>
  <span
    class="icon-wrapper"
  />
</button>
`;

exports[`menu-bar-item people matches snapshot 1`] = `
<button
  class="item item-people"
  type="button"
>
  <span
    class="icon-wrapper"
  />
</button>
`;

exports[`menu-bar-item search matches snapshot 1`] = `
<button
  class="item item-search"
  type="button"
>
  <span
    class="icon-wrapper"
  />
</button>
`;

exports[`menu-bar-item toggle matches snapshot 1`] = `
<button
  class="item item-toggle"
  type="button"
>
  <span
    class="icon-wrapper"
  />
</button>
`;

exports[`menu-bar-item when all state, all item has highlight matches snapshot 1`] = `
<button
  class="item item-all active"
  type="button"
>
  <span
    class="icon-wrapper"
  />
</button>
`;
